### DISCUSSION TITLE
Issues with uploads using Cloudflare tunnel

### DISCUSSION THREAD

**Discussion by paxiongmao** [LycheeOrg/Lychee#2265] [Q&A] [answered]
Hi All,
Apologies if this is the wrong place for this discussion. I am running Lychee via Portainer / Docker and am using a Cloudflare Tunnel to expose the gallery to the wider world. I'm using the lycheeorg/lyuchee:latest image (5.1.1).

When accessing the instance using an internal IP (192.168.1.x:90) everything works fine. When I change the APP_URL to my domain and redeploy, I can access the gallery (https://gallery.my.domain/gallery), but cannot upload. I get the message `"Upload failed. The server returned an error!"`. Looking at my browser console I see the following error: `Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist.`

Any help would be much appreciated, I assume there's something else in my config that I need to change, or possibly something with the http/https setup on the tunnel. Diagnostic info below:

    Diagnostics
    -----------
        Warning: Dropbox import not working. dropbox_key is empty.
        Info: Latest version of PHP is 8.3
        Warning: git (software) is not available.
     

    System Information
    ------------------
        Lychee Version (git):                    ?? (1e3b000) -- Could not compare.
        DB Version:                              5.1.1
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               false
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.2.7
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                Europe/London
        Max uploaded file size:                  100M
        Max post size:                           100M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      3600
        MySQL Version:                           10.11.6-MariaDB-1:10.11.6+maria~ubu2204
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1691
        GD Version:                              2.3.3
        Number of foreign key:                   12 found.
     
 
Thanks in advance for any help or pointers.

**Comment by ildyria** [comment, marked as answer]
Did you set up `TRUSTED_PROXIES` ?

**Reply by relink2013** [reply]
I don't know if this helps in this case or not, but CF tunnels have a 100MB file upload limit. This may be what you're hitting and it's dropping the connection. 

The only way around this would be to implement chunked uploading like several other apps have had to do.

**Reply by ildyria** [reply]
He is using the Livewire front-end so that 100MB upload file limit is no longer applicable, we now support the chunked uploading. :)

**Comment by paxiongmao** [comment]
Hi All,
Thanks for the suggestions. I added the config updates in this issue: (https://github.com/LycheeOrg/Lychee-Docker/issues/175) for trusted proxies and it worked.

For info I didn't need anything on the Cloudflare side to get this working. 

Thank you very much for the prompt replies!

**Comment by paxiongmao** [comment]
Added to the compose file (Portainer stack definition):

    TRUSTED_PROXIES: *
    APP_URL: https://a.b.com:3444
    APP_FORCE_HTTPS: true
